File: cpp/Bit Manipulation/swap_odd_and_even_bits.cpp
unsigned int swapOddAndEvenBits(unsigned int n) {
    unsigned int evenMask = 0x55555555; // 01010101010101010101010101010101
    unsigned int oddMask  = 0xAAAAAAAA; // 10101010101010101010101010101010
    unsigned int evenBits = n & evenMask;
    unsigned int oddBits  = n & oddMask;
    // Shift the even bits to the left, the odd bits to the right, and
    // merge these shifted values together.
    return (evenBits << 1) | (oddBits >> 1);
}

File: java/Bit Manipulation/SwapOddAndEvenBits.java
public class SwapOddAndEvenBits {
    public int swapOddAndEvenBits(int n) {
        int evenMask = 0x55555555; // 01010101010101010101010101010101
        int oddMask = 0xAAAAAAAA;  // 10101010101010101010101010101010
        int evenBits = n & evenMask;
        int oddBits = n & oddMask;
        // Shift the even bits to the left, the odd bits to the right, and 
        // merge these shifted values together.
        return (evenBits << 1) | (oddBits >> 1);
    }
}
